lib/email_alert_signup_api.rb

```diff
@@ -1,23 +1,33 @@
 require 'email_alert_title_builder'
+require 'validate_query'
 
 class EmailAlertSignupAPI
   def initialize(dependencies = {})
     @email_alert_api = dependencies.fetch(:email_alert_api)
     @attributes = dependencies.fetch(:attributes)
     @subscription_list_title_prefix = dependencies.fetch(:subscription_list_title_prefix)
     @available_choices = dependencies.fetch(:available_choices)
+    @default_attributes = dependencies.fetch(:default_attributes, {})
   end
 
   def signup_url
     subscriber_list['subscription_url']
   end
 
+  def validate!
```

Comment:
This adds an extra call to rummager. Is this necessary? - we could let email-alert-api reject invalid input?

Reply:
Yeah I agree. Do you think it's OK to leave it in here, and remove it once we've added validation to email-alert-api?